plugins/jobs/src/js/components/JobDeleteModal.js

```diff
@@ -0,0 +1,77 @@
+import * as React from "react";
+import { Confirm } from "reactjs-components";
+import StringUtil from "#SRC/js/utils/StringUtil";
+import UserActions from "#SRC/js/constants/UserActions";
+
+function defaultContent(jobId) {
+  const defaultMessage = `Are you sure you want to ${
+    UserActions.DELETE
+  } ${jobId}? This action is irreversible.`;
+
+  const defaultLabel = `${StringUtil.capitalize(UserActions.DELETE)} Job`;
+
+  return {
+    label: defaultLabel,
+    message: modalContentHtml(defaultMessage)
+  };
+}
+
+function stopJobsCotent(jobId) {
+  const stopCurrentRunsLabel = `Stop Current Runs and ${StringUtil.capitalize(
+    UserActions.DELETE
+  )} Job`;
+
+  const stopCurrentRunsMessage = `Couldn't ${
+    UserActions.DELETE
+  } ${jobId} as there are currently active job runs. Do you want to stop all runs and ${
+    UserActions.DELETE
+  } the job?`;
+
+  return {
+    label: stopCurrentRunsLabel,
+    message: modalContentHtml(stopCurrentRunsMessage)
+  };
+}
+
+function modalContent(jobId, stopCurrentJobRuns) {
```

Comment:
This is one of the rare cases where I think that having these put into functions makes it more unreadable and harder to understand. Can we either do an if on the call-side or get rid of this builder pattern?

Reply:
I agree with you here and the HTML function. I did the `quick extraction ™`